app/src/main/java/org/mozilla/fenix/components/metrics/AppLaunchTimeMeasurement.kt

```diff
@@ -0,0 +1,59 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.components.metrics
+
+import android.os.Process
+import android.os.SystemClock
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import org.mozilla.fenix.perf.Stat
+
+/**
+ * Handles the logic of figuring out launch time for cold, warm and hot startup.
+ */
+class AppLaunchTimeMeasurement(private val stats: Stat = Stat()) {
+
+    // when user launched the time. i.e press app_icon, launched a link.
+    private var processLaunchTimeStampNanoTime: Long? = null
+    // we are considering screen to be visible after the first pre draw call.
+    private var elapsedNanoTimestampOnHomeActivityPreDraw: Long? = null
+
+    fun onApplicationOnCreate() {
+        // There is disk read involve and we dont want to block the main thread.
+        GlobalScope.launch(Dispatchers.IO) {
+            processLaunchTimeStampNanoTime = stats.getProcessStartTimeStampNano(Process.myPid())
+        }
+    }
+
+    fun onHomeActivityOnCreate() {
+        if (processLaunchTimeStampNanoTime == null) {
+            // if timestamp was not set, this is likely a warm
+            processLaunchTimeStampNanoTime = homeActivityInitTimeStampNano
```

Comment:
I think the comment is a bit vague (could do without I think). Imo,  since this function only serves one purpose, maybe name it to be more self-explanatory such as `onHomeActivityOnCreateIsColdStart` (or something a big easier to understand).

Reply:
Hmm good point. I can make it more clear? This function will be called for both cold and warm startup so I was having a hard time to find a good name.